src/Queues/DbQueue.php

```diff
@@ -0,0 +1,202 @@
+     * CREATE TABLE queue (
+     *     id BIGINT NOT NULL PRIMARY KEY AUTO_INCREMENT
+     *     status TINYINT NOT NULL DEFAULT 0
+     *     timestamp DATETIME NOT NULL DEFAULT CURRENT_TIMESTAMP
+     *     data LONGBLOB
+     * );
+     * ```
+     * @var string
+     */
+    public $tableName = '{{%queue}}';
+    
+    /**
+     * Whether to do hard delete of the deleted job, instead of just flagging the
+     * status.
+     * @var boolean
+     */
+    public $hardDelete = true;
+    
+    /**
+     * @return void
+     */
+    public function init()
+    {
+        parent::init();
+        $this->db = \yii\di\Instance::ensure($this->db, \yii\db\Connection::className());
+    }
+    
+    /**
+     * Return next job from the queue.
+     * @return Job|boolean the job or false if not found.
+     */
+    protected function fetchJob()
+    {
+        //Avoiding multiple job.
+        $transaction = $this->db->beginTransaction();
+        $row = $this->fetchLatestRow();
+        if ($row == false || !$this->flagRunningRow($row)) {
+            $transaction->rollBack();
+            return false;
+        }
+        $transaction->commit();
+        
+        $job = $this->deserialize($row['data']);
+        $job->id = $row['id'];
+        $job->header['timestamp'] = $row['timestamp'];
+        
+        return $job;
+    }
+    
+    /**
+     * Fetch latest ready job from the table.
+     *
+     * Due to the use of AUTO_INCREMENT ID, this will fetch the job with the
+     * largest ID.
+     *
+     * @return array
+     */
+    protected function fetchLatestRow()
+    {
+        return (new \yii\db\Query())
+                    ->select('*')
+                    ->from($this->tableName)
+                    ->where(['status' => self::STATUS_READY])
+                    ->orderBy(['id' => SORT_DESC])
+                    ->limit(1)
+                    ->one($this->db);
+    }
+    
+    /**
+     * Flag a row as running. This will update the row ID and status if ready.
+     *
+     * @param array $row The row to update.
+     * @return boolean Whether successful or not.
+     */
+    protected function flagRunningRow(array $row)
+    {
+        $updated = $this->db->createCommand()
+                ->update(
+                    $this->tableName,
+                    ['status' => self::STATUS_ACTIVE],
+                    [
+                        'id' => $row['id'],
+                        'status' => self::STATUS_READY,
+                    ]
+                )->execute();
+        return $updated == 1;
+    }
+
+    /**
+     * Post new job to the queue.  This contains implementation for database.
+     *
+     * @param Job $job The job to post.
+     * @return boolean whether operation succeed.
+     */
+    protected function postJob(Job $job)
+    {
+        return $this->db->createCommand()->insert($this->tableName, [
+            'timestamp' => new \yii\db\Expression('NOW()'),
+            'data' => $this->serialize($job),
```

Comment:
use column name as attribute, so people who already have existing database doesn't need to alter their table.